conan/tools/cmake/cmakedeps/templates/config.py

```diff
@@ -67,11 +68,17 @@ def template(self):
             endif()
         endforeach()
 
-        set({{ file_name }}_VERSION_STRING "{{ version }}")
-        set({{ file_name }}_INCLUDE_DIRS {{ pkg_var(pkg_name, 'INCLUDE_DIRS', config_suffix) }} )
-        set({{ file_name }}_INCLUDE_DIR {{ pkg_var(pkg_name, 'INCLUDE_DIRS', config_suffix) }} )
-        set({{ file_name }}_LIBRARIES {{ pkg_var(pkg_name, 'LIBRARIES', config_suffix) }} )
-        set({{ file_name }}_DEFINITIONS {{ pkg_var(pkg_name, 'DEFINITIONS', config_suffix) }} )
+        {% if legacy_variable_prefix %}
+            {% set prefix_name = legacy_variable_prefix %}
+        {% else %}
+            {% set prefix_name = file_name %}
+        {% endif %}
+
+        set({{ prefix_name }}_VERSION_STRING "{{ version }}")
+        set({{ prefix_name }}_INCLUDE_DIRS {{ pkg_var(pkg_name, 'INCLUDE_DIRS', config_suffix) }} )
+        set({{ prefix_name }}_INCLUDE_DIR {{ pkg_var(pkg_name, 'INCLUDE_DIRS', config_suffix) }} )
+        set({{ prefix_name }}_LIBRARIES {{ pkg_var(pkg_name, 'LIBRARIES', config_suffix) }} )
+        set({{ prefix_name }}_DEFINITIONS {{ pkg_var(pkg_name, 'DEFINITIONS', config_suffix) }} )
```

Comment:
Doing the change I suggested you can leave this as it was but changing `file_name` to `legacy_variable_prefix`